### PR TITLE
fix(wasm): Avoid requesting stdlib paths from the source-resolver

### LINKED ISSUE #1244: Dependencies in noir wasm don't get std library
### Aim

To compile a dependency with noir_wasm that uses the std library.

### Expected behavior

The dependency should have access to the std library.

### Bug

Instead, the compiler crashes 
```
panicked at 'error reporter: could not find crate std', crates/noirc_frontend/src/hir/resolution/import.rs:199:28

Stack:

Error
    at module.exports.__wbg_new_abda76e883ba8a5f (/mnt/user-data/alvaro/aztec3-packages/yarn-project/node_modules/@noir-lang/noir_wasm/nodejs/noir_wasm.js:212:17)
    at wasm://wasm/00f971da:wasm-function[1910]:0x2c0557
    at wasm://wasm/00f971da:wasm-function[1210]:0x2a05ef
    at wasm://wasm/00f971da:wasm-function[1577]:0x2b8472
    at wasm://wasm/00f971da:wasm-function[235]:0x188092
    at wasm://wasm/00f971da:wasm-function[1478]:0x2ae78e
    at wasm://wasm/00f971da:wasm-function[283]:0x198208
    at wasm://wasm/00f971da:wasm-function[29]:0x6185a
    at wasm://wasm/00f971da:wasm-function[29]:0x5f05c
    at wasm://wasm/00f971da:wasm-function[29]:0x5ef2b
```

### To reproduce

1. write a resolver that allows to resolve dependencies
2. create a library that uses the std lib
3. compiler will crash

### Installation method

None

### Nargo version

_No response_

### @noir-lang/noir_wasm version

_No response_

### @noir-lang/barretenberg version

_No response_

### @noir-lang/aztec_backend version

_No response_

### Additional context

_No response_

### Submission Checklist

- [ ] Once I hit submit, I will assign this issue to the Project Board with the appropriate tags.

### PR DESCRIPTION
**Pull request by phated**
# Description

## Problem\*

Resolves #1244 

## Summary\*

This fixes the noir_wasm integration tests, such that they succeed. The fundamental problem was that the implementation of `source-resolver` isn't actually checking file paths to know what files to load. Instead, it was always returning the entry file as the source code, so this meant the stdlib was setup all wrong inside the compiler.

While this is a lazy way to implement the source-resolver, none of the stdlib file paths should be reaching it in the first place. So I've added an extra case in the `file_reader.rs` to never go to the filesystem for the stdlib, which we always embed.

Also note that some stuff in the integration test needed to change because proving takes 6.5 minutes and verifying takes 5 minutes.

## Documentation

- [ ] This PR requires documentation updates when merged.

  

  - [ ] I will submit a noir-lang/docs PR.

  

  - [ ] I will request for and support Dev Rel's help in documenting this PR.

  
  

## Additional Context

# PR Checklist\*

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: compiler/integration-tests/test/integration/browser/compile_prove_verify.test.ts [hunk 1]
```diff
@@ -49,11 +49,11 @@ const numberOfThreads = navigator.hardwareConcurrency || 1;
 
 let suite = Mocha.Suite.create(mocha.suite, "Noir end to end test");
 
-suite.timeout(60*10e3);//10mins
+suite.timeout(60*20e3);//20mins
```
Comment (kevaundray) [inline compiler/integration-tests/test/integration/browser/compile_prove_verify.test.ts:52]: oof

File: compiler/integration-tests/test/integration/browser/compile_prove_verify.test.ts [hunk 2]
```diff
@@ -132,14 +132,15 @@ test_cases.forEach((testInfo) => {
 
             const acirComposer = await api.acirNewAcirComposer(CIRCUIT_SIZE);
 
+            // This took ~6.5 minutes!
             const proof = await api.acirCreateProof(
```
Comment (kevaundray) [inline compiler/integration-tests/test/integration/browser/compile_prove_verify.test.ts:136]: 6.5 minutes is quite a long time, perhaps this is running in single threaded mode

Comment (kevaundray) [reply compiler/integration-tests/test/integration/browser/compile_prove_verify.test.ts:136]: In the aztec-packages repo which also has tests that mirror these, it takes on the order of seconds, so definitely suspicious

Comment (kevaundray) [reply compiler/integration-tests/test/integration/browser/compile_prove_verify.test.ts:136]: This is orthogonal and was already there so not blocking

**Review by kevaundray — APPROVED** [review]
LGTM - so if it cannot find the stdlib assets and the path is the path to the stdlib ie something that begins with `std/` then it errors. This is better than erroneously looking at the file system for something which is not there

**Review by TomAFrench — APPROVED** [review]
LGTM

**Review by kobyhallx — DISMISSED** [review]
LGTM

**Review by TomAFrench — CHANGES_REQUESTED** [review]
CI is failing

**Comment by phated** [comment]
> CI is failing

This is the wasm test that constantly times out. It hasn't been working for weeks and is not a required check that blocks PRs.

**Review by kobyhallx — APPROVED** [review]

**Pull request merged**